### ISSUE TITLE
Updated Seurat Object (v2) can't be plotted in Seurat v3 DimPlot

### ISSUE THREAD

**Issue by rlorenzc** [satijalab/seurat#1509]
Hi,

I am trying to plot an updated object with Seurat v3's DimPlot function and I'm getting this error back:

> DimPlot(object = o1_updated, dims = "umap", group.by = "sample")

Error in DimPlot(object = o1_updated, dims = "umap", group.by = "sample") : 
  'dims' must be a two-length vector

And when I look into umap in the reductions slot:
> o1_updated@reductions$umap
A dimensional reduction object with key UMAP_ 
 Number of dimensions: 2 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: RNA 

Any help would be appreciated!

Thanks!

**Comment by timoast** [comment]
Hi, the `dims` argument refers to which dimensions to plot, not which dimension reduction method, and should be a numeric vector of length two. To change the dimension reduction method use `reduction`. See the documentation for `DimPlot` for more information.

**Comment by rlorenzc** [comment]
Sorry, that was my mistake! 

This is my problem now when I run the following:
DimPlot(object = o1_updated, reduction = "umap", group.by = "sample")

Error in `[.data.frame`(data, , c(dims, x, split.by, shape.by)) : undefined columns selected

**Comment by rlorenzc** [comment]
I also get the same problem when I run FeaturePlot:

Error in `[.data.frame`(data, , dims[1]) : undefined columns selected

**Comment by rlorenzc** [comment]
We figured out the problem. When the object is being updated from v2 to v3, and you used UMAP, they colnames for the cell.embeddings is "UMAP1" and "UMAP2." And v3 looks for "UMAP_1", and "UMAP_2, " so simply renaming the colnames fixes the problem. 

This naming issue is not present with tSNE.

**Comment by MichaelStber** [comment]
Dear @rlorenzc ,

May I ask you how do you renamed the colnames to UMAP_1 and UMAP_2???

I tried this but its not working:

colnames(Embeddings(object = myobject, reduction = "umap"))[1] <- "UMAP_1"

**Comment by MichaelStber** [comment]
dear @rlorenzc 
dear @timoast 

I tried now sveeral ways but I was not able to change the column names. I am very sorry for this banal thing, but I guess I need help.

They can also be listed with this command:
>colnames(myobject[["umap"]])
[1] "UMAP1" "UMAP2"

I tried so svereal ways to rename it but it didn’t work:

>myobject <- colnames(myobject[["umap"]]) <- c("UMAP_1", "UMAP_2") 
Error in dimnames(x) <- dn : 'dimnames' applied to non-array

>myobject <- colnames(myobject[["umap"]])[1] <- "UMAP_1" 
Error in dimnames(x) <- dn : 'dimnames' applied to non-array

>myobject <- colnames(myobject[["umap"]]) %>% 
  rename(
    UMAP1 = UMAP_1,
    UMAP2 = UMAP_2
  )
Error in UseMethod("rename_") : 
  no applicable method for 'rename_' applied to an object of class "character"

Does anyone has an idea??
I would be really glad for help.
Thank you.

Best wishes
